src/main/java/org/edumips64/MainCLI.java

```diff
@@ -62,146 +67,31 @@ public static void main(String args[]) {
-            }
-          }
-        } else if (tokens[0].compareTo("run") == 0) {
-            int steps = 0;
-            long startTimeMs = System.currentTimeMillis();
-            try {
-              while(true) {
-                steps++;
-                c.step();
-
-              }
-            } catch (HaltException e) {
-              long endTimeMs = System.currentTimeMillis();
-              long totalTimeMs = endTimeMs - startTimeMs;
-              System.out.println("Execution ended. " + steps + " steps were executed in " + totalTimeMs + "ms");
-            }
-        } else if (tokens[0].compareTo("step") == 0) {
-          try {
-            int num = 1;
-
-            if (tokens.length > 1) {
-              num = Integer.parseInt(tokens[1]);
-            }
-
-            try {
-              if (num > 0) {
-                System.out.println("I execute " + num + " steps of simulation.");
-              }
+      runReplLoop(commandLine);
 
-              for (int i = 0; i < num; ++i) {
-                c.step();
-                System.out.println(c.pipeLineString());
-              }
-            } catch (Exception e) {
-              System.err.println(getErrorReportingMessage());
-              e.printStackTrace();
-            }
-          } catch (NumberFormatException e) {
-            System.out.println("The second parameter of the step command must be an integer.");
-          }
-        } else {
-          System.out.println("Unknown command.\nType 'help' to get the list of available commands.");
-        }
-
-        System.out.print("> ");
-      }
-
-      System.out.println("Bye!");
     } catch (Exception e) {
-      System.err.println(getErrorReportingMessage());
-      e.printStackTrace();
+      Cli.printErrorMessage(e);
       System.exit(1);
     }
   }
-  
-  /**
-   * Compiles and returns a generic error message with version 
-   * and system diagnostic information to be used to report the error.
-   * @return A string containing the error message.
-   */
-  private static String getErrorReportingMessage() {
-    String msg = "EduMIPS64 fatal error!\n" +
-        "Please report the following stacktrace and system information,\n" +
-        "along with the content of the assembly file you were executing\n" +
-        "to the EduMIPS64 GitHub account: https://github.com/lupino3/edumips64/issues/new\n";
-    msg += String.format("Version: %s, %s, %s\n", MetaInfo.VERSION, MetaInfo.BUILD_DATE, MetaInfo.FULL_BUILDSTRING);
-    msg += String.format("JRE version: %s\nOS: %s\n\n", System.getProperty("java.version"), System.getProperty("os.name"));
-    return msg;
-  }  
+
+    private static void runReplLoop(CommandLine commandLine) throws IOException {
+        BufferedReader keyboard = new BufferedReader(new InputStreamReader(System.in));
+        System.out.print("> ");
+        while (true) {
+          String read = keyboard.readLine();
```

Comment:
It might be worth to not invoke `execute` if the input is empty or contains only whitespace, as that will return to the user a non-informative user message (`Unmatched argument at index 0: ''`)

Reply:
I agree. I would think that printing the help message would be ideal for entering no input, thoughts?

Reply:
Yeah, makes sense! Thanks!